### PR TITLE
fix(mapping): Synthesize filenames for Kotlin classes buried in synthetic wrappers

### PR DESCRIPTION
**Pull request by romtsn**
Fix synthesized source filenames for Compose-compiler-generated `ComposableSingletons$FileKt` classes when they are inlined through R8 synthetic lambdas.

When a Compose compilation emits a `ComposableSingletons$MainActivityKt` class, the mapping often does not include a top-level declaration for it — only its `$$ExternalSyntheticLambda*` siblings appear, each carrying `sourceFile: R8$$SyntheticClass`. In that case the outer Kotlin class has no `sourceFile` metadata of its own, and frame resolution falls back to `synthesize_source_file`, which previously did `split('$').next()` and collapsed `ComposableSingletons$MainActivityKt` to `ComposableSingletons`. The resulting frame's filename became `ComposableSingletons.java`, dropping the real `MainActivity.kt`.

The fix scans every `$`-separated segment of the last dotted component for a Kotlin file-class marker (ends in `Kt`). The inner `MainActivityKt` segment now wins and produces `MainActivity.kt`. Bare top-level Kotlin classes (`FooKt`) and non-Kotlin inner classes (`Foo$Bar` → `Foo.java`) are unchanged — they still resolve on the first iteration exactly as before.

Includes a minimal reproducer in `tests/r8-source-file-edge-cases.rs` mirroring the scenario: a `ComposableSingletons$MyScreenKt` lambda inlined through an `$$ExternalSyntheticLambda` whose `sourceFile` is `R8$$SyntheticClass`.

### CODE REVIEW

**Review by Dav1dde — APPROVED** [review]

File: src/utils.rs [hunk 1]
```diff
@@ -13,17 +13,22 @@ pub(crate) fn extract_class_name(full_path: &str) -> Option<&str> {
 /// For example: ("com.example.Main", Some("Other.kt")) -> "Main.kt"
 /// For example: ("com.example.MainKt", None) -> "Main.kt"
 /// For inner classes: ("com.example.Main$Inner", None) -> "Main.java"
+/// For Compose-wrapped Kotlin files: ("com.example.ComposableSingletons$MainKt", None) -> "Main.kt"
```
Comment (loewenheim) [inline src/utils.rs]: If you wanted to be fancy, all of this could be a doctest.

File: tests/r8-source-file-edge-cases.rs [hunk 2]
```diff
@@ -117,6 +117,36 @@ fn test_source_file_name_synthesize_stacktrace() {
...
+// =============================================================================
+// ComposableSingletons inlined lambda
+//
+// When a lambda from a Compose `ComposableSingletons$FileKt` wrapper is inlined
+// through an R8 synthetic class, the outer class has no `sourceFile` metadata of
+// its own (only its `$$ExternalSyntheticLambda*` children appear in the mapping,
+// carrying `R8$$SyntheticClass`). The synthesized filename must be driven by the
+// inner `FooKt` segment, not by stripping after the first `$`.
+// =============================================================================
+
```
Comment (loewenheim) [inline tests/r8-source-file-edge-cases.rs]: IMO make this a doc comment on the constant.
```suggestion
/// ComposableSingletons inlined lambda
///
/// When a lambda from a Compose `ComposableSingletons$FileKt` wrapper is inlined
/// through an R8 synthetic class, the outer class has no `sourceFile` metadata of
/// its own (only its `$$ExternalSyntheticLambda*` children appear in the mapping,
/// carrying `R8$$SyntheticClass`). The synthesized filename must be driven by the
/// inner `FooKt` segment, not by stripping after the first `$`.
```

**Review by loewenheim — CHANGES_REQUESTED** [review]

File: src/utils.rs (hunk 1 above)
Comment (romtsn) [reply src/utils.rs]: I think doctest does not cover `pub(crate)` methods, so I converted those into real unit-tests, if that's fine?

Comment (loewenheim) [reply src/utils.rs]: Ah, you're right, that annoys me every time. Thanks for adding the tests though!

**Review by loewenheim — APPROVED** [review]
